buildchain/buildchain/ui.py

```diff
@@ -70,4 +73,22 @@ def clean() -> None:
     }
 
 
+def task__ui_config() -> types.TaskDict:
+    """Copy MetalK8s UI configuration to the build directory"""
+    source = constants.ROOT/'images'/'metalk8s-ui'/'conf'/'nginx.conf'
+    target = config.BUILD_ROOT/'metalk8s-ui-nginx.conf'
+
+    def clean(target: Path) -> Callable[[], None]:
+        return lambda: coreutils.rm_rf(target)
+
+    return {
+        'title': utils.title_with_target1('COPY'),
+        'actions': [(coreutils.cp_file, (source, target))],
+        'targets': [target],
+        'task_dep': ['_build_root'],
+        'file_dep': [source],
+        'clean': [clean(target)],
```

Comment:
want to delete `target`, then you can simply use `'clean': True`.

> A common scenario is a task that needs to “revert” its actions. A task may include a clean attribute. This attribute can be True to remove all of its target files.

See https://pydoit.org/cmd_other.html#clean

You only need a custom callable when you want to delete stuff that aren't in `targets`.

Reply:
Nice. Will do.